etchosts/etchosts.go

```diff
@@ -119,25 +119,34 @@ func Add(path string, recs []Record) error {
 		return nil
 	}
 
-	f, err := os.Open(path)
+	b, err := mergeRecords(path, recs)
```

Comment:
It is only refactoring to scope the close to happen before the write occurs, since it is open read only it is not too important since there is no sync that needs to occur.